lib/ui/namespaces.js

```diff
@@ -32,12 +43,104 @@ function namespaces_list(screen) {
     }
   });
 
-  return namespaces_list;
+  blessed.text({
+    parent  : namespaces_box,
+    width   : 9,
+    left    : 3,
+    top     : 1,
+    align   : 'left',
+    content : 'Filter :'
+  }); 
+
+  const search = blessed.textbox({
+    parent       : namespaces_box,
+    border       : 'line',
+    width        : '85%',
+    height       : 3,
+    top          : 0,
+    right        : 1,
+    keys         : true,
+    mouse        : true,
+    inputOnFocus : true,
+    style : {
+      fg       : 'white',
+      label    : { bold: true },
+      border   : { fg: 'white' },
+      selected : { bg: 'blue' },
+    }
+  });
+
+  // Canonical way of extending components in Blessed
+  search.__oolistener = search._listener;
+  search._listener = function(ch, key) {
+    const word = ['up', 'down','enter'];    
+    if (word.includes(key.name)) {
+        return namespaces_list.emit('keypress', ch, key);
+    }
+    const ret = this.__oolistener(ch, key);
+    let i = 0;
+    namespaces_list.items.forEach((element, pos) => {
+        if (!element.getContent().includes(search.value)){
+            element.position.top = 0;
+            element.hidden = true;
+        } else {
+            element.position.top = i;
+            element.hidden = false;
+            if (i===0) {
+                namespaces_list.selected = pos;
+            }
+            i++;
+        }
+    });
+    namespaces_list.scrollTo(0);
+    screen.render();
+    return ret;
+  };
+  namespaces_box.on('keypress', (ch, key) => {
+    const keys = ['escape', 'up', 'down', 'enter', 'q', 'left', 'right'];
+    if (keys.includes(key.name)) {
+        namespaces_list.emit('keypress', ch, key);
+    } else {
+        screen.saveFocus();
+        search.focus();
+        search.readInput();
+        search._listener(ch,key);
+        screen.render();
+    } 
+  });
+
+  namespaces_list.up = function(offset) {
+    return move(offset, (offset) => (offset || 1) * -1, (i, len) => mod(i-1, len));
+  };
+  
+  namespaces_list.down = function(offset) {
+    return move(offset, (offset) => (offset || 1), (i, len) => mod(i+1, len));
+  };
+
+  function move(offset, start, inc) {
+    if (namespaces_list.items[namespaces_list.selected].hidden)
+      return;
+    let i = namespaces_list.selected + start(offset);
+    const len = namespaces_list.items.length;
+    for (i = mod(i,len); i < len && i >= 0 ; i = inc(i, len)) {
+        if (!namespaces_list.items[i].hidden) { 
+            namespaces_list.selected = i;
+            namespaces_list.scrollTo(namespaces_list.items[i].position.top);
+            screen.render();
+            return;
+        }
+    }
+  }
+
+  function mod(a, b) {
+    return (((a)%b)+b)%b;
+  }
+  return { 'search': search, 'box': namespaces_box, 'list': namespaces_list };
```

Comment:
Can be replaced with `{ search, box, list } `.

Reply:
Replaced